plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ChooseOpenshiftConnectionComposite.java

```diff
@@ -177,8 +177,14 @@ public void widgetSelected(SelectionEvent e) {
 			}
 		});
 		if (connectionCombo.getItemCount() > 0) {
-			connectionCombo.select(0);
-			setSelectedConnection(connections.get(0));
+			IConnection recentConnection = ConnectionsRegistrySingleton.getInstance().getRecentConnection();
+			for(String connectionItem : connectionCombo.getItems()) {
+				if(connectionItem.contains(recentConnection.getHost())) {
+					connectionCombo.select(connectionCombo.indexOf(connectionItem));
+					break;
+				}
+			}
+			setSelectedConnection(recentConnection);
```

Comment:
I would suggest to extract l180-187 to a separate method and improve readability and maybe reuse it. 
Another suggestion then is to use a ```ComboViewer``` instead of a ```Combo``` to eliminate all the item/connection matching operations that one can find in this class.
Such duplicate and unnecessary code is ex. at
* l101
* l116
* l181
* l219